File: punt.js
/*
  classe Punt 3D
  - no té cap dependència
*/

class Punt{
  constructor(x,y,z){
    this.x=x||0;//coordenada x
    this.y=y||0;//coordenada y
    this.z=z||0;//coordenada z
  }

  //calcula distància entre 2 punts
  distancia(punt){
    let x=punt.x-this.x;
    let y=punt.y-this.y;
    let z=punt.z-this.z;
    return Math.sqrt(x*x + y*y + z*z);
  }
}

module.exports=Punt;

/*
test:
  let p=new Punt(3,4,12);
  console.log(p);
  //-> Punt { x: 3, y: 4, z: 12 }
  console.log(p.distancia(new Punt(0,0,0)))
  //-> 13
  */

File: structs/Punt.c
/*punt.c*/
#include "punt.h"

//mostra info objecte punt
void print_punt(Punt *p){
  //mostra coordenades i lloc memòria
  printf("Punt{x:%lf, y:%lf, z:%lf} @%p\n",
    p->x, p->y, p->z, p);

}

//calcula distància entre 2 punts
double distancia(Punt *p1, Punt *p2){
  double x = p1->x - p2->x;
  double y = p1->y - p2->y;
  double z = p1->z - p2->z;
  return sqrt(x*x + y*y + z*z);
};
